exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,3034 @@
+			if err != nil {
+				close(tpfChan)
+				log.Fatal("failed to format exchange currency for options pair")
+			}
+		}
+		if d.CurrencyPairs.IsAssetEnabled(asset.OptionCombo) == nil {
+			tradablePair, err = d.GetEnabledPairs(asset.OptionCombo)
+			if err != nil {
+				close(tpfChan)
+				log.Fatalf("failed to update tradable pairs. Err: %v", err)
+			} else if len(tradablePair) == 0 {
+				close(tpfChan)
+				log.Fatalf("enabled %v for asset type %v", currency.ErrCurrencyPairsEmpty, asset.OptionCombo)
+			}
+			optionComboTradablePair = tradablePair[0]
+		}
+		if d.CurrencyPairs.IsAssetEnabled(asset.FutureCombo) == nil {
+			tradablePair, err = d.GetEnabledPairs(asset.FutureCombo)
+			if err != nil {
+				close(tpfChan)
+				log.Fatalf("failed to update tradable pairs. Err: %v", err)
+			} else if len(tradablePair) == 0 {
+				close(tpfChan)
+				log.Fatalf("enabled %v for asset type %v", currency.ErrCurrencyPairsEmpty, asset.FutureCombo)
+			}
+			futureComboTradablePair = tradablePair[0]
+		}
+		tradablePairsFetchedStatusLock.Lock()
+		tradablePairsFetched = true
+		tradablePairsFetchedStatusLock.Unlock()
+		close(tpfChan)
+	}(fetchTradablePairChan)
+}
+
+func sleepUntilTradablePairsUpdated() {
+	tradablePairsFetchedStatusLock.Lock()
+	if tradablePairsFetched {
+		tradablePairsFetchedStatusLock.Unlock()
+		return
+	}
+	tradablePairsFetchedStatusLock.Unlock()
+	<-fetchTradablePairChan
+}
+
+func TestGetLatestFundingRates(t *testing.T) {
+	t.Parallel()
+	_, err := d.GetLatestFundingRates(context.Background(), &fundingrate.LatestRateRequest{
+		Asset:                asset.USDTMarginedFutures,
+		Pair:                 currency.NewPair(currency.BTC, currency.USDT),
+		IncludePredictedRate: true,
+	})
+	require.ErrorIs(t, err, asset.ErrNotSupported)
+	result, err := d.GetLatestFundingRates(context.Background(), &fundingrate.LatestRateRequest{
+		Asset: asset.Futures,
+		Pair:  futuresTradablePair,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, result)
+}
+
+func TestUpdateOrderExecutionLimits(t *testing.T) {
+	t.Parallel()
+	err := d.UpdateOrderExecutionLimits(context.Background(), asset.Spot)
+	require.NoErrorf(t, err, "Error fetching %s pairs for test: %v", asset.Spot, err)
+	instrumentInfo, err := d.GetInstruments(context.Background(), "BTC", d.GetAssetKind(asset.Spot), false)
+	require.NoError(t, err)
+	require.NotEmpty(t, instrumentInfo, "invalid instrument information found")
+	limits, err := d.GetOrderExecutionLimits(asset.Spot, spotTradablePair)
+	require.NoErrorf(t, err, "Asset: %s Pair: %s Err: %v", asset.Spot, spotTradablePair, err)
+	require.Equalf(t, instrumentInfo[0].TickSize, limits.PriceStepIncrementSize, "Asset: %s Pair: %s Expected: %v Got: %v", asset.Spot, spotTradablePair, instrumentInfo[0].TickSize, limits.MinimumBaseAmount)
+	assert.Equalf(t, instrumentInfo[0].MinimumTradeAmount, limits.MinimumBaseAmount, "Pair: %s Expected: %v Got: %v", spotTradablePair, instrumentInfo[0].MinimumTradeAmount, limits.MinimumBaseAmount)
```

Comment:
```
    deribit_test.go:2842: 
        	Error Trace:	/Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/deribit/deribit_test.go:2842
        	Error:      	Not equal: 
        	            	expected: 0.01
        	            	actual  : 1
        	Test:       	TestUpdateOrderExecutionLimits
        	Messages:   	Asset: spot Pair: BTC_USDC Expected: 0.01 Got: 0.0001
```

Reply:
Why am I not getting this error?

Reply:
This is a result from `testnet` being `true` so my mistake in not saying so.

But, this test fails there because of assumptions in the test: ` instrumentInfo[0]` there is no check or guarantee that it will be `spotTradablePair`. It would be an improvement to make sure that that `instrument[x]` is equal to `spotTradablePair` before compairing results. This will help protect it against future changes on the server API

Reply:
I see. Where this is coming from?  I have added a filter to select the target instrument. Thank you!